src/Grpc.AspNetCore.Server/Internal/ServerCallHandlerBase.cs

```diff
@@ -34,7 +34,7 @@ protected ServerCallHandlerBase(Method<TRequest, TResponse> method, GrpcServiceO
         {
             Method = method;
             ServiceOptions = serviceOptions;
-            Logger = loggerFactory.CreateLogger(GetType());
```

Comment:
Is this an optimization?

Reply:
It is so the error is related to the user's service implementation type.